core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -267,6 +267,10 @@ private TableProperties() {
   public static final String METADATA_DELETE_AFTER_COMMIT_ENABLED = "write.metadata.delete-after-commit.enabled";
   public static final boolean METADATA_DELETE_AFTER_COMMIT_ENABLED_DEFAULT = false;
 
+  public static final String METRICS_MAX_INFERRED_COLUMN_DEFAULTS =
```

Comment:
nit: `INFERRED` doesn't really make sense to me in this context.  We're not inferring, we're actually explicit here.

Reply:
I debated what to call this. The problem is that we are defaulting the default. You can explicitly set a default using `write.metadata.metrics.default`, and that will apply to all columns. If you don't _set_ a default we _infer_ one for you, but only for the first 32 columns (at least, after this PR). That's why I used "inferred" -- I thought it was better than `max-defaulted-default-columns`.

Reply:
Agree with this table property, initially I had made one but it was taken out during the discussions.  Indeed it's a bit of a confusing config, but I dont see any other great option.

Reply:
The problem with "persist" and similar is that it misses the distinction between an explicit default (when `write.metadata.metrics.default` is set) and an implicit default that comes from Iceberg. I think the right behavior is to preserve what we currently do, which is to use the explicit default for all columns. But that means that this property should obviously not apply to the explicit default. That's why I used "inferred default".

What about changing this to include `unconfigured` or `missing`? Something like `missing-mode-limit`?

Reply:
Let's just keep it as-is then. I don't think it is a big deal.